aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,798 @@
+
+    Args:
+        ec2_client: aws boto client
+        security_grp_id (str): The ID of Security Group to add inbound rule with
+        sg_rule (dict): The dictionary containing of security inbound rule
+
+    Raises:
+        ClientError: If error response is provided by an AWS service to Boto3 client’s request.
+    """
+    try:
+        ec2_client.authorize_security_group_ingress(
+            GroupId=security_grp_id,
+            CidrIp=sg_rule["CidrIp"],
+            IpProtocol=sg_rule["IpProtocol"],
+            FromPort=sg_rule["FromPort"],
+            ToPort=sg_rule["ToPort"],
+        )
+    except ClientError as ec2_client_err:
+        LOGGER.error(ec2_client_err)
+        raise
+
+
+def ec2_add_outbound_security_rule(ec2_client, security_grp_id, sg_rule) -> None:
+    """
+    Adds Outbound Rule to security group
+
+    Args:
+        ec2_client: aws boto client
+        security_grp_id (str): The ID of Security Group to add outbound rule with
+        sg_rule (dict): The dictionary containing of security outbound rule
+
+    Raises:
+        ClientError: If error response is provided by an AWS service to Boto3 client’s request.
+    """
+    try:
+        ec2_client.authorize_security_group_egress(
+            GroupId=security_grp_id,
+            CidrIp=sg_rule["CidrIp"],
+            IpProtocol=sg_rule["IpProtocol"],
+            FromPort=sg_rule["FromPort"],
+            ToPort=sg_rule["ToPort"],
+        )
+    except ClientError as ec2_client_err:
+        LOGGER.error(ec2_client_err)
+        raise
+
+
+def ec2_add_security_rules(ec2_client, security_grp_id, rules_json_file) -> None:
+    """
+    Adds Inbound and Outbound Rules to security group
+
+    Args:
+        ec2_client: aws boto client
+        security_grp_id (str): The ID of Security Group to add inbound/outbound rule with
+        rules_json_file (str): The file name of security inbound and outbound rules in json
+
+    Raises:
+        ClientError: If error response is provided by an AWS service to Boto3 client’s request.
+    """
+    with open(rules_json_file) as rules_file:
+        rules_object = json.load(rules_file)
+
+    try:
+        if "inbound_rules" in rules_object.keys():
+            for rule in rules_object["inbound_rules"]:
+                ec2_add_inbound_security_rule(ec2_client, security_grp_id, rule)
+
+        if "outbound_rules" in rules_object.keys():
+            for rule in rules_object["outbound_rules"]:
+                ec2_add_outbound_security_rule(ec2_client, security_grp_id, rule)
+
+    except ClientError as ec2_client_err:
+        LOGGER.error(ec2_client_err)
+        raise
+
+
+def create_public_subnets_for_vpc(
+    ec2_client, subnet_count, vpc_id, subnet_cidr_blocks, availability_zones, site_name
+) -> list:
+    """
+    creates public subnets in a given vpc
+
+    Args:
+         ec2_client: aws boto client
+         subnet_count(int): The number of additional bits with which to extend the prefix
```

Comment:
please fix the docstring in the code
`allocate_elastic_ips(int)`  >> `allocate_elastic_ips (int)` << missing space before (type)

Reply:
done